src/lib/gssapi/spnego/gssapiP_spnego.h

```diff
@@ -12,7 +12,12 @@
 extern "C" {
 #endif
 
+typedef struct spnego_ctx_st *spnego_gss_ctx_id_t;
```

Comment:
Why did this definition move?

Reply:
The NegoEx function prototypes refer to spnego_gss_ctx_id_t, but the SPNEGO types refer to NegoEx types.  So right now gssapiP_negoex.h has to be included in a particular place, after spnego_gss_ctx_id_t is declared but before the SPNEGO types are declared.  It's possible that switching to `struct negoex_auth_mech` instead of a typedef will allow a less awkward coupling; I will check.

Reply:
> It's possible that switching to struct negoex_auth_mech instead of a typedef will allow a less awkward coupling; I will check.

The remaining sticking point is "conversation_id negoex_conv_id;" in the SPNEGO context.  I'm not sure there's a good way to finesse that.

(There are of course ways around this.  The NegoEx stuff could be placed in a separate, dynamically allocated sub-context.  Or we could split up the NegoEx type declarations from the NegoEx function prototypes so they're in separate files.  Or we could combine everything into gssapiP_spnego.h.  I'm just not sure if any of those options is better than the current awkwardness where gssapiP_negoex.h has to be included from gssapiP_spnego.h in a specific place.)

Reply:
Thank you for clarifying.  Mostly it just wasn't immediately obvious why it was happening.